unix: fix assertion failure when the signal pipe overflows

piscisaureus opened the pull request:
An incorrect assert() statement was causing libuv to crash when writing
to an internal signal pipe would result in EAGAIN/EWOULDBLOCK.

This commit doesn't solve the underlying issue that the signal pipe can
overflow.

This should fix joyent/node#5538

Nodejs-Jenkins commented:
Thank you for contributing this pull request! Here are a few pointers to make sure your submission will be considered for inclusion.

Commit joyent/libuv@9df9b5d1e86f7cea7c0ab3e73ce54ae1db3fc7c1 has the following error(s):
- First line of commit message must be no longer than 50 characters

Please see [CONTRIBUTING.md](https://github.com/joyent/node/blob/master/CONTRIBUTING.md) for more information

piscisaureus committed:
unix: fix assert on signal pipe overflow

An incorrect assert() statement was causing libuv to crash when writing
to an internal signal pipe would result in EAGAIN/EWOULDBLOCK.

This commit doesn't solve the underlying issue that the signal pipe can
overflow.

This should fix joyent/node#5538
